### DISCUSSION TITLE
GitHub Action needed: GitHub Personal Access Token found in gist, any app using this secret may be affected

### DISCUSSION THREAD

**Discussion by Jemt** [community/community#134685] [Code Security]
### Select Topic Area

Question

### Body

Hi,

Today I received the following e-mail message from GitHub:

```
Your GitHub Personal Access Token has been revoked

A recent scan found a valid GitHub Personal Access Token, <identifier which looks familiar to me>, linked to your GitHub account in the content of gist (link to: https://gist.github.com/z580975290857612847918/42cd9ddef7d0d936912820b6f027b9a3/d38c13a626d5a842078160a88fba4b1d32a1a35b#file-0______Learn_more_at__watchtower.billdemirkapi.me_____.txt). We have revoked it to protect your data from unauthorized access, and as a consequence, any app using this token won't be able to authenticate to GitHub. 
```

I seriously doubt that I have posted a Personal Access Token on Gist, and notice the very odd URL to the gist which contains "Learn more at _watchtower dot billdemirkapi dot me_". The link to the gist just shows a "404 not found" error.

The e-mail does look legit though, and Github's security log shows the following two entries from today:

```
- Removed authorization for personal access token (<identifier which looks familiar to me>)
- Revoked a Personal access token (Classic) token for <repository> ending in ********.
```

I would really like to know more about this non-existing gist and the personal access token. I have scanned all my servers and workstations for the Personal Access Token ending in ******* (`grep -i ****** -R *`) but found nothing.

Is this perhaps an error caused by Githubs security scanner ?

**Comment by Jemt** [comment]
Turns out [watchtower.billdemirkapi.me](https://watchtower.billdemirkapi.me/) actually provided some useful details.
A security scanner detected an access token, and reported it to GitHub which automatically revoked it. I suspect they removed the gist as well.

To mitigate any potential security threats, I reviewed all commits made in the period after creating the access token, enabled GitHub's [code security scanner](https://docs.github.com/en/code-security/code-scanning/introduction-to-code-scanning/about-code-scanning) (which is free for open source projects), including Secret Scanning and Push Protection, and configured signing of commits to avoid the risk of anyone else committing on my behalf, even if they have my credentials, without it being immediately noticeable. I recommend 2FA as well.

I'm a bit disappointed with the level of information provided by GitHub. For instance, I don't remember the access levels for this access token, and as it has been automatically revoked and removed, I'm unable to determine the actual risk it posed. I would also like to know the content of the gist in question - was it just an unlikely match or an attempt to construct malicious code? I am grateful that access was revoked automatically though.

**Comment by automainint** [comment]
Same thing happened to me just now.

**Comment by ghostinhershell** [comment]
Hi @Jemt, 👋🏻 We really appreciate you flagging this. The best route to get this to the proper GitHub team is to use our [abuse reporting tools](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam). Here's all the info:

- [Reporting abuse or spam](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam)
- [Reporting a user](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-user)
- [Reporting an issue or pull request](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-an-issue-or-pull-request)
- [Reporting a comment](https://docs.github.com/en/communities/maintaining-your-safety-on-github/reporting-abuse-or-spam#reporting-a-comment)

You can report behavior and content that violates community guidelines and terms. We are going to close this post, but for this and any future incidents, please refer to the links above.

Thank you!
